lib/action/InvokeAPIController.dart

```diff
@@ -76,6 +80,27 @@ class InvokeAPIController {
     }
   }
 
+  void _updateAPIState(
+    ScopeManager? scopeManager,
+    InvokeAPIAction action, {
+    required bool isLoading,
+    required bool isSuccess,
+    required bool isError,
+  }) {
+    // Setting Success
+    dispatchAPIChanges(
```

Comment:
This might not work. If the API was called previously it would have existing data. Upon calling it again, the data will be wiped out for a brief moment, then populated with the new data. This is not what we want.

Data should stay intact until the new data overrides it. The bindings will jerk (disappear then re-appearing while the API is being refetched)

Reply:
Can you check now? I updated the PR. I'm replacing the API Response with the same data and with new state (ex: loading or error)